core/primitives-core/src/hash.rs

```diff
@@ -134,9 +150,9 @@ impl Hash for CryptoHash {
 /// let data = [1, 2, 3];
 /// let hash = near_primitives_core::hash::hash(&data);
 /// ```
+#[deprecated(since = "#5252", note = "Use CryptoHash::hash_bytes or ::hash_borsh")]
```

Comment:
I think I was confusing here. I do think that the APIs in this PR with inherent functions on `CrytpoHash` are abstractly better and more idiomatic. I don't think they are significantly better enough to justify the change. Even if we do change them, I feel that's better done in a follow up PR.

Just marking the method as deprecated I think *adds* to technical debt, instead of removing it, as it moves codebase from "consistent, but not ideal" to "inconsistent, mostly not ideal with small islands which are better".

Reply:
My plan is to ship PR's to make the change. At the very least do in it places where where using `try_to_vec`, or some other memory allocation is not optimal.

I didn't do this in this PR. To keep things simple, and I kind of expected that we may change the api anyway.

I'm fine with removing the `#[deprecated(since = "#5252", note = "Use CryptoHash::hash_bytes or ::hash_borsh")]`

Reply:
Ok, I removed deprecated.

My goal right now is to clean up `near-network` code. We probably should clean up code in other packages as well, but that can be done later.